src/core/lib/layer.js

```diff
@@ -578,6 +579,17 @@ export default class Layer {
     }
   }
 
+  _deleteResources(newState) {
```

Comment:
You may want to also check `state,models`

Don't we want to do this in `finalizeLayer` as well?